### PR TITLE
add AgentResolver

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

File: resolvers/agent.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		return out, errors.WithStack(err)
+	}
+	defer func() {
+		err2 := resp.Body.Close()
+		if err != nil {
+			err = errors.WithStack(err2)
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
```
Comment (ilya-korotya) [inline resolvers/agent.go:109]: Accept range of 2xx codes.

File: resolvers/agent.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+// NewAgentResolver returns new agent resolver
+func NewAgentResolver(config AgentResolverConfig) *AgentResolver {
+	return &AgentResolver{config}
+}
+
+// Resolve is a method to resolve a credential status from an agent.
+func (r AgentResolver) Resolve(ctx context.Context,
+	status verifiable.CredentialStatus) (out verifiable.RevocationStatus, err error) {
+
```
Comment (ilya-korotya) [inline resolvers/agent.go:54]: return an error if status.Type != agetStatusType

**Review by ilya-korotya — CHANGES_REQUESTED** [review]

File: resolvers/agent.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+func GetSenderDID(ctx context.Context) *w3c.DID {
+	v := ctx.Value(ctxKeySenderDID{})
+	if v == nil {
+		return nil
+	}
+	return v.(*w3c.DID)
+}
+
+// AgentResolverConfig options for credential status verification
+type AgentResolverConfig struct {
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply resolvers/agent.go:37]: no, it looks like you missed our discussions

File: resolvers/agent.go (hunk 2 above)
Comment (vmidyllic) [reply resolvers/agent.go:54]: +1

File: resolvers/agent.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+	revocationBody := protocol.RevocationStatusRequestMessageBody{
+		RevocationNonce: status.RevocationNonce,
+	}
+	rawBody, err := json.Marshal(revocationBody)
+	if err != nil {
+		return out, errors.WithStack(err)
+	}
+
+	idUUID, err := uuid.NewV4()
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply resolvers/agent.go:63]: +1

File: resolvers/agent.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+		return out, errors.WithStack(err)
+	}
+
+	resp, err := http.DefaultClient.Post(status.ID, "application/json", bytes.NewBuffer(iden3commMsg))
+	if err != nil {
+		return out, errors.WithStack(err)
+	}
+	defer func() {
+		err2 := resp.Body.Close()
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply resolvers/agent.go:104]: just return revocationStatus.RevocationStatus, err in the end of the function

File: resolvers/agent.go (hunk 1 above)
Comment (vmidyllic) [reply resolvers/agent.go:109]: not sure why

Comment (vmidyllic) [reply resolvers/agent.go:109]: 200 is expected on all other clients.

**Review by vmidyllic — COMMENTED** [review]
some comments from Illiya are ok

File: resolvers/agent.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply resolvers/agent.go]: done

File: resolvers/agent.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+	if err != nil {
+		return out, errors.WithStack(err)
+	}
+
+	iden3commMsg, err := r.config.PackageManager.Pack(packers.MediaTypePlainMessage, bytesMsg, nil)
+	if err != nil {
+		return out, errors.WithStack(err)
+	}
+
+	resp, err := http.DefaultClient.Post(status.ID, "application/json", bytes.NewBuffer(iden3commMsg))
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply resolvers/agent.go]: done

File: resolvers/agent_test.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	senderDID, err := w3c.ParseDID("did:polygonid:polygon:mumbai:2qFDziX3k3h7To2jDJbQiXFtcozbgSNNvQpb6TgtPE")
+	require.NoError(t, err)
+
+	agentConfig := AgentResolverConfig{
+		PackageManager: pm,
+	}
+
+	agentResolver := NewAgentResolver(agentConfig)
+
+	httpmock.Activate()
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply resolvers/agent_test.go]: done

File: resolvers/agent.go (hunk 2 above)
Comment (volodymyr-basiuk) [reply resolvers/agent.go:55]: done

File: resolvers/agent.go (hunk 1 above)
Comment (volodymyr-basiuk) [reply resolvers/agent.go]: done

**Review by ilya-korotya — APPROVED** [review]

File: go.mod [hunk 8]
```diff
@@ -5,12 +5,13 @@ go 1.18
 require (
 	github.com/dustinxie/ecc v0.0.0-20210511000915-959544187564
 	github.com/gofrs/uuid/v5 v5.0.0
+	github.com/google/uuid v1.6.0
```
Comment (vmidyllic) [inline go.mod:8]: why do we have to uuid packages? I know Illiya aksed to use goohle uuid, but in this case let's remove gofrs completely.

Comment (volodymyr-basiuk) [reply go.mod]: removed

**Review by vmidyllic — APPROVED** [review]

**Review by ilya-korotya — APPROVED** [review]

**Pull request merged**